src/wallet/wallet.h

```diff
@@ -365,82 +365,24 @@ struct COutputEntry
     int vout;
 };
 
-/** A transaction with a merkle branch linking it to the block chain. */
+/** Legacy class used for deserializing vtxPrev for backwards compatibility **/
 class CMerkleTx
 {
-private:
-  /** Constant used in hashBlock to indicate tx has been abandoned */
-    static const uint256 ABANDON_HASH;
-
 public:
-    CTransactionRef tx;
-    uint256 hashBlock;
-
-    /* An nIndex == -1 means that hashBlock (in nonzero) refers to the earliest
-     * block in the chain we know this or any in-wallet dependency conflicts
-     * with. Older clients interpret nIndex == -1 as unconfirmed for backward
-     * compatibility.
-     */
-    int nIndex;
-
-    CMerkleTx()
-    {
-        SetTx(MakeTransactionRef());
-        Init();
-    }
-
-    explicit CMerkleTx(CTransactionRef arg)
-    {
-        SetTx(std::move(arg));
-        Init();
-    }
-
-    void Init()
-    {
-        hashBlock = uint256();
-        nIndex = -1;
-    }
-
-    void SetTx(CTransactionRef arg)
-    {
-        tx = std::move(arg);
-    }
-
     ADD_SERIALIZE_METHODS;
 
     template <typename Stream, typename Operation>
     inline void SerializationOp(Stream& s, Operation ser_action) {
+        CTransactionRef tx;
+        uint256 hashBlock;
+        int nIndex;
```

Comment:
Would this write uninitialized memory to the wallet file?

Reply:
No `CMerkleTx` object is ever serialized.

I could go even further in this PR and remove the `CMerkleTx` object entirely. Let me add a commit to see what that looks like.

Reply:
Yeah, I am interested in seeing that commit. That probably means you can get rid of the `static_cast<const CMerkleTx&>(*this)` that happens in the serialization of `CWalletTx`

Reply:
> `static_cast<const CMerkleTx&>(*this)`

Oh, that probably means that serialization *is* called, so I'd like to re-raise my previous question of uninitialized memory.

Reply:
> That probably means you can get rid of the static_cast<const CMerkleTx&>(*this) that happens in the serialization of CWalletTx

Huh? This PR already does get rid of the `static_cast<const CMerkleTx&>(*this)`

Reply:
I like the assert

Reply:
> But the following removes the serialization support so maybe not worth it?

I agree (if we decide to keep the second commit)